R/run_filter.R

```diff
@@ -0,0 +1,144 @@
+#' @description
+#' Computes unique groups
+#' @param min_count_run filter parameter. 
+#' @param min_pres Run filter parameter. The minimum proportion of presence in the time period for a series of signals grouped
+#'  by m/z to be considered a peak.
+#' @param profile The matrix containing m/z, retention time, intensity, and EIC label as columns.
+#' @return unique_grp. 
+compute_uniq_grp <- function(profile, min_count_run, min_pres = 0.6) {
+  grps <- profile  
+  ttt <- table(grps)
+  ttt <- ttt[ttt >= max(min_count_run * min_pres, 2)]
+  unique_grp <- as.numeric(names(ttt))
+  return(unique_grp)
+}
+
+#' @description
+#' Computes the smoothed retention times by using The Nadaraya-Watson kernel regression estimate function.
+#' @param min_run Run filter parameter. The minimum length of elution time for a series of signals grouped by m/z to be considered a peak.
+#' @param times. Retention times vector.
+#' @return predicted rt.
+#' @examples
+#' predict_smoothed_rt(min_run = min_run, times)
+predict_smoothed_rt <- function(min_run = 5, times) {
+  # ksmooth(x, y, kernel, bandwidth, range, n.points, x.points)
+  smooth <- ksmooth(
+    seq(-min_run + 1, length(times) + min_run), 
+    c(rep(0, min_run),   
+    times,       
+    rep(0, min_run)),    
+    kernel = "box",      
+    bandwidth = min_run, 
+    x.points = 1:length(times) 
+  ) 
+  # vector of smoothed estimates for the regression at the corresponding x
+  smooth <- smooth$y  
+  return(smooth)
+}
+
+#' @description
+#' This function labels the indices of values kept to perform further calculations
+#' @param min_run Run filter parameter. The minimum length of elution time for a series of signals grouped by m/z to be considered a peak.
+#' @param min_pres Run filter parameter. The minimum proportion of presence in the time period for a series of signals grouped
+#'  by m/z to be considered a peak.
+#' @param timeline. 
+#' @param this_times. 
+#' @param times. Retention times vector.
+#' @return to_keep. 
+label_val_to_keep <- function(min_run, timeline, min_pres, this_times, times) {
+    this_timeline <- timeline
+    this_timeline[this_times] <- 1
+    to_keep <- this_times * 0
+    
+    # filtering based on the kernel regression estimate
+    this_smooth <- predict_smoothed_rt(min_run, this_timeline)
+    if (max(this_smooth) >= min_pres) {
+      measured_points <- good_points <- timeline
+      measured_points[this_times] <- 1
+
+      good_sel <- which(this_smooth >= min_pres)
+      good_points[good_sel] <- 1
+      for (j in (-min_run):min_run)
+      {
+        curr_sel <- good_sel + j
+        curr_sel <- curr_sel[curr_sel > 0 & curr_sel <= length(times)]
+        good_points[curr_sel] <- 1
+      }
```

Comment:
Just a minor style thing - consider keeping the curly brackets consistent (not just here). Maybe once you resolve the conflicts, R lint might have a comment on that (I'm not sure).

```suggestion
    if (max(this_smooth) >= min_pres) {
      measured_points <- good_points <- timeline
      measured_points[this_times] <- 1

      good_sel <- which(this_smooth >= min_pres)
      good_points[good_sel] <- 1
      for (j in (-min_run):min_run) {
        curr_sel <- good_sel + j
        curr_sel <- curr_sel[curr_sel > 0 & curr_sel <= length(times)]
        good_points[curr_sel] <- 1
      }
```